cellbender/remove_background/vae/encoder.py

```diff
@@ -107,184 +128,207 @@ def forward(self, x: torch.Tensor, **kwargs) -> Dict[str, torch.Tensor]:
-
-
-class EncodeNonEmptyDropletLogitProb(nn.Module):
-    """Encoder module that transforms gene expression into cell probabilities.
-
-    The number of input units is the total number of genes and the number of
-    output units is the dimension of the latent space.  This encoder transforms
-    a point in high-dimensional gene expression space to a latent probability
-    that a given barcode contains a real cell, via a learned transformation
-    involving fully-connected layers.  This encoder uses both the gene
-    expression counts as well as an estimate of the ambient RNA profile in
-    order to output a cell probability.
-
-    Args:
-        input_dim: Number of genes.  The size of the input of this encoder.
-        hidden_dims: Size of each of the hidden layers.
-        output_dim: Number of dimensions of the probability estimate (1).
-        input_transform: Name of transformation to be applied to the input
-            gene expression counts.  Must be one of
-            ['log', 'normalize', 'log_center'].
         log_count_crossover: The log of the number of counts where the
             transition from cells to empty droplets is expected to occur.
-
-    Attributes:
-        transform: Name of transformation to be applied to the input gene
-            expression counts.
-        param: The log of the number of counts where the transition from
-            cells to empty droplets is expected to occur.
         linears: torch.nn.ModuleList of fully-connected layers before the
             output layer.
         output: torch.nn.Linear fully-connected output layer for the size
             of each input barcode.
-        input_dim: Size of input gene expression.
+        n_genes: Size of input gene expression.
 
-    Note:
+    Returns:
+        output: Dict containing -
+            logit_p: Logit probability that droplet contains a cell
+            d: Cell size scale factor
+            epsilon: Value near one that represents droplet RT efficiency
+
+    Notes:
         An encoder with two hidden layers with sizes 100 and 500, respectively,
         should set hidden_dims = [100, 500].  An encoder with only one hidden
         layer should still pass in hidden_dims as a list, for example,
         hidden_dims = [500].
-        The output is in the form of a logit, so can be any real number.  The
-        transformation from logit to probability is a sigmoid.
+        The output is in the form of a dict.  Ouput for cell probability is a
+        logit, so can be any real number.  The transformation from logit to
+        probability is a sigmoid.
+        Several heuristics are used to try to encourage a good initialization.
 
     """
 
-    def __init__(self, input_dim: int, hidden_dims: List[int], output_dim: int,
-                 log_count_crossover: float, input_transform: str = None):
-        super(EncodeNonEmptyDropletLogitProb, self).__init__()
-        self.input_dim = input_dim
+    def __init__(self,
```

Comment:
please refactor all weight initializations out of the `__init__` and to a private method called by `__init__`.

Reply:
Sounds good